photutils/daogroup.py

```diff
@@ -0,0 +1,138 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Functions for implementing the DAOGROUP algorithm proposed by Stetson 1987.
+"""
+
+from __future__ import division
+import numpy as np
+from astropy.table import Column, Table, vstack
+
+__all__ = ['daogroup']
+
+def daogroup(starlist, crit_separation):
+    """
+    This is an implementation which follows the DAO GROUP algorithm presented
+    by Stetson (1987).
+
+    GROUP divides an entire starlist into sets of distinct, self-contained
+    groups of mutually overlapping stars.
+
+    GROUP accepts as input a list of stars and their approximate
+    brightenesses relative to a model stellar intensity profile for the frame
+    and determines which stars are close enough to be capable of
+    adversely influencing each others' profile fits.
+
+    Parameters
+    ----------
+    starlist : `~astropy.table.Table` or array-like
+        List of stars positions.
+        If `~astropy.table.Table`, columns should be named 'x_0' and 'y_0'.
+        Additionally, 'flux_0' may also be provided. 
+        TODO: If array-like, it should be either (x_0, y_0) or
+        (x_0, y_0, flux_0).
+        If 'starlist' only contains x_0 and y_0, 'crit_separation' must be
+        provided.
+    crit_separation : float
+        Distance, in units of pixels, such that any two stars separated by
+        less than this distance will be placed in the same group.
+        TODO: If None, 'flux_0' must be provided in 'starlist'.
+    
+    Returns
+    -------
+    group_starlist : list of `~astropy.table.Table`
+        Each `~astropy.table.Table` in the list corresponds to a group of
+        mutually overlapping starts.
+
+    Notes
+    -----
+    Assuming the psf fwhm to be known, 'crit_separation' may be set to
+    k*fwhm, for some positive real k.
+
+    See
+    ---
+    `~daofind`
+    """
+
+    if not isinstance(crit_separation, type(1.0)):
```

Comment:
also, since it's in pixel, we may want to accept ints, too? 
That case you can do a try except, with  something along the line ``crit_separation = float(crit_separation)``

Though this is not important, and I'm not sure how much slower it would be.